dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -37,46 +48,87 @@
 
     private static final Logger log = LoggerFactory.getLogger(RetryingChannel.class);
 
+    /*
+     * Shared single thread executor is reused between all retrying channels. If it becomes oversaturated
+     * we may wait longer than expected before resuming requests, but this is an
+     * edge case where services are already operating in a degraded state and we should not
+     * spam servers.
+     */
+    private static final Supplier<ListeningScheduledExecutorService> sharedScheduler = Suppliers.memoize(
+            () -> MoreExecutors.listeningDecorator(Tracers.wrap(
+                    "dialogue-RetryingChannel-scheduler",
+                    Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
+                            .setNameFormat("dialogue-RetryingChannel-scheduler-%d")
+                            .setDaemon(false)
+                            .build()))));
```

Comment:
one more flup please- can we have instrumentation on this? e.g. how many things are currently on it? or will there be a way that I can pass a WC scheduled executor to this? Also possibly want an uncaught exception handler given the recent spiciness we discovered

Reply:
good idea, would be nice to get metrics on this and identical tracing to other components.